src/TSTransformer/nodes/transformOptionalChain.ts

```diff
@@ -257,10 +259,13 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
+					const [index] = state.capture(() => transformExpression(state, item.argumentExpression));
```

Comment:
Is there a reason you're capturing the prereqs here? And then discarding them?

Reply:
i forgot to remove that 
oops